### PR TITLE
changing ip address to string for localhost

### PR DESCRIPTION
**Pull request by infinite-Joy**
For https://github.com/pallets/flask/issues/2007

which now is more inline with docs in the home page of http://flask.pocoo.org/

Before the change:
```
python hello_flask.py
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [07/Sep/2016 11:08:55] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [07/Sep/2016 11:08:55] "GET /favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [07/Sep/2016 11:09:14] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [07/Sep/2016 11:09:14] "GET /favicon.ico HTTP/1.1" 404 -
```

After the change:
```
python hello_flask.py
 * Running on http://localhost:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [07/Sep/2016 11:09:42] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [07/Sep/2016 11:09:59] "GET / HTTP/1.1" 200 -
```

As you can see its not "getting" the favicon. Is that an issue?

---
This change is [Reviewable](https://reviewable.io/reviews/pallets/flask/2010)

### CODE REVIEW

**Comment by ThiefMaster** [comment]
> As you can see its not "getting" the favicon. Is that an issue?

Probably just means your browser already has one cached.

File: flask/app.py [hunk 1]
```diff
@@ -827,7 +829,7 @@ def run(self, host=None, port=None, debug=None, **options):
         """
         from werkzeug.serving import run_simple
         if host is None:
-            host = '127.0.0.1'
+            host = 'localhost'
```
Comment (ThiefMaster) [inline flask/app.py]: This is not equivalent to `127.0.0.1` - using `localhost` may result in `::1` being used instead isince `localhost` has both IPv4 and IPv6 addresses

**Comment by davidism** [comment]
Please remove the unrelated grammar "fixes", they're not necessary.

**Comment by infinite-Joy** [comment]
done

**Comment by davidism** [comment]
From the linked issue:

> I think we should only allow `None` and `!= 'localhost'`.

So what exactly is this addressing by making it equal `localhost`?

**Comment by untitaker** [comment]
I think y'all are misunderstanding #2007. I want to show an error when the user tries to set `app.config['SERVER_NAME']` to an IP address.

**Pull request closed without merging**